Automattic/wp-calypso
# Fix email forward notices for wrong domains

#### Changes proposed in this Pull Request

* This change introduces a new `EmailExistingForwardsNotice` component based on the existing `TitanExistingForwardsNotice` component that can be used in various contexts, but especially from the Email Comparison page.
* The new component only shows a warning when the current domain has email forwards
* We now show this new component in the Email Comparison page and the account creation page for Google
* We no longer show this component when adding mailboxes for Professional Email or Google

#### Testing instructions

* Ensure that you have a site with at least three domains, one of which is configured with at least one email forward, one which has G Suite or Google Workspace, and one that has Professional Email.
* Visit the page to add a mailbox for the Professional Email account (`/email/:domainName/titan/new/:siteSlug` or Upgrades -> Emails -> :domainName -> Add new mailboxes)
* Verify that we don't show the email forward notice for this domain
* Visit the page to add a mailbox for the G Suite/Google Workspace account (`/email/:domainName/gsuite/add-users/:siteSlug` or Upgrades -> Emails -> :domainName -> Add new mailboxes)
* Verify that we don't show the email forward notice for this domain
* Visit the page for creating a new Google Workspace account, but make sure you use the domain name for the domain that has email forwarding (`/email/:domainName/google-workspace/new/:siteSlug`)
* Verify that you *do* see the email forward notice on the page and that the notice mentions the correct domain name
* Visit the email comparison page for the domain that has email forwarding (`/email/:domainName/purchase/:siteSlug`)
* Verify that you *do* see the email forward notice on the page and that the notice mentions the correct domain name

#### Screenshots

<img width="1057" alt="Screenshot 2021-06-22 at 17 03 16" src="https://user-images.githubusercontent.com/3376401/122999097-d2712280-d37b-11eb-8b92-62cbafc38899.png">
<img width="1057" alt="Screenshot 2021-06-22 at 16 58 59" src="https://user-images.githubusercontent.com/3376401/122999098-d309b900-d37b-11eb-8f3e-189f2ebe77ec.png">


Related to #53307
